Src/IronPython/Runtime/PythonEncoding.cs

```diff
@@ -85,22 +93,48 @@ public PythonEncoding(Encoding encoding, PythonEncoderFallback encoderFallback,
             Pass2Encoding.DecoderFallback = decoderFallback;
         }
 
-        public override int GetByteCount(char[] chars, int index, int count) {
+        private void PrepareResidentEncoder() {
```

Comment:
It is a very nice idea. I have tried it out locally and indeed it works with `Nullable` at `1.3.0` and `LangVersion` at `preview`. But do we really want to set the language version to `preview`? As I understand it, it allows the code to use any language features from C# 9.0 without warnings, but some features from above C# 7.3 may not work on .NET Framework or .NET Core 2.x. If we accidentally use any of them, we loose compiler errors.

Anyway, there are more places that may benefit from `MemberNotNullAttibute` (e.g. I'm thinking of `MemoryView`), so I think it is better to do in a separate PR.

Reply:
On the other hand, I tried to "accidentally" use a few of those newer features and the build still fails for one reason or another. So maybe my concern is not justified.

Reply:
Yeah, I'm not too concerned about it, as you say if it's supported then it'll work, otherwise it'll fail in one way or another. We could also wait for the November release (that way we don't have to roll back `preview` to `latest`). Just don't want to forget about it.

Reply:
I'd rather wait will November. Who knows what surprises a "preview" version may have. I have set a reminder so no worry about forgetting.